awscli/customizations/s3/s3handler.py

```diff
@@ -367,169 +381,125 @@ def _enqueue_upload_end_task(self, filename, upload_context):
         self.executor.submit(complete_multipart_upload_task)
 
 
-class S3StreamHandler(S3Handler):
+class S3TransferStreamHandler(BaseS3Handler):
     """
     This class is an alternative ``S3Handler`` to be used when the operation
     involves a stream since the logic is different when uploading and
     downloading streams.
     """
-    # This ensures that the number of multipart chunks waiting in the
-    # executor queue and in the threads is limited.
-    MAX_EXECUTOR_QUEUE_SIZE = 2
-    EXECUTOR_NUM_THREADS = 6
 
     def __init__(self, session, params, result_queue=None,
                  runtime_config=None):
-        if runtime_config is None:
-            # Rather than using the .defaults(), streaming
-            # has different default values so that it does not
-            # consume large amounts of memory.
-            runtime_config = RuntimeConfig().build_config(
-                max_queue_size=self.MAX_EXECUTOR_QUEUE_SIZE,
-                max_concurrent_requests=self.EXECUTOR_NUM_THREADS)
-        super(S3StreamHandler, self).__init__(session, params, result_queue,
-                                              runtime_config)
+        super(S3TransferStreamHandler, self).__init__(
+            session, params, result_queue, runtime_config)
+        self.config = create_transfer_config_from_runtime_config(
+            self._runtime_config)
+
+        # Restrict the maximum chunks to 1 per thread.
+        self.config.max_in_memory_upload_chunks = \
```

Comment:
I am really confused as to why that test fails for one over the other then. Because if we are matching the number of threads (which is 10), the number of in memory chunks should be 10. However if we uncouple the two, the number of in memory chunks is still 10... Why is that? Or am I off in my logic somewhere?